### ISSUE TITLE
Documentation?

### ISSUE THREAD

**Issue by Grawl** [Nipher/gulp-npm-files#2]
It's hard to understand what does boolean argument

```js
  gulp.src(gnf(true), {base:'./'}).pipe(gulp.dest('./build'));
--------------------^ here
```

**Comment by goodseller** [comment]
In the `readme.md`:
```
    // Copy dependencies and devDependencies to build/node_modules/
```
So it's about loading `dependencies` and `devDependencies` in npm file to array and for gulp pipe to use.

**Comment by Grawl** [comment]
Okay. It's just hard to learn by code comments only.

**Comment by Nipher** [comment]
Maybe we could modify this to 
```javascript
gulp.src(gnf({ dev: true }), {base:'./'}).pipe(gulp.dest('./build'));
```

what do you guys think?
@goodseller @Grawl

**Comment by Grawl** [comment]
@Nipher I think it will be better with

```js
gulp.src(gnf('devDependencies'), {base:'./'}).pipe(gulp.dest('./build'));
```

or

```js
gulp.src(gnf('optionalDependencies'), {base:'./'}).pipe(gulp.dest('./build'));
```

Just use the key from `package.json` directly :wink:

**Comment by goodseller** [comment]
@Nipher @Grawl 

I do not have strong opinion on the way of handling it, in either using associated array or array. The idea are both good on it's own.

I have had a quick lookup from the [offical site](https://docs.npmjs.com/files/package.json#dependencies). And found these way of handling dependencies:
 - dependencies
 - devDependencies
 - peerDependencies
 - bundledDependencies / bundleDependencies
 - optionalDependencies

It might need to be aware that for [bundledDependencies](https://docs.npmjs.com/files/package.json#bundleddependencies):

    If this is spelled "bundleDependencies", then that is also honored.


**In short**, Do we need to handle it?

If so use `{}` is better. (npm file path can be also include in it)
If not, `[]` is good enough and can handle the need that if more kind of dependencies will be added to npm file in the future.



**Last but not least**, is it nice to keep the default value of `gnf()` as loading `dependencies`?

Thanks.

**Comment by Grawl** [comment]
@goodseller 

> **In short**, Do we need to handle it?

Do you about to handle both bundleDepencencies and bundle**d**Dependencies?

I think if `npm` handle it so we need to handle it.

> **Last but not least**, is it nice to keep the default value of `gnf()` as loading `dependencies`?

Yes.

If we provide an object of settings for `gnf()` so we need a default object with just `target: 'dependencies'` for example.

If you want to add some things into config, I prefer to use an object. Like this:

```js
gnf({
  target: [
    `dependencies`,
    `devDependencies`
  ],
  package: './path/to/package.json'
});
```

Note that `target` can handle both `string` and `array` (`string` is just an array of 1 entity to ease of use).

*You can use different word than `target`, it's for example only.*

**Comment by goodseller** [comment]
@Grawl That looks good and is ok for me. :+1: Good Idea for the object, it has both of the benefits.

**Comment by Nipher** [comment]
That's sounds really good!
I like it a lot!
Thanks guys, I will implement this as soon as I can!
or if you guys prefer, you can do it too :smile:

**Comment by Grawl** [comment]
@Nipher I cannot write logic, sorry

**Comment by Nipher** [comment]
@Grawl No problem! 
thank you for sharing your ideas for the plugin!
I will try to do this as soon as I can

**Comment by Grawl** [comment]
@Nipher :sunglasses:
